### PR TITLE
Support llama3.1-8b generation

### PR DESCRIPTION
**Pull request by Gasoonjia**
Llama3.1 8b now is supported in torchchat! 🎉

Local test:
image

### CODE REVIEW

**Review by Jack-Khuu — APPROVED** [review]

File: build/model.py [hunk 1]
```diff
@@ -361,15 +365,41 @@ def forward(self, x: Tensor) -> Tensor:
...
+    scale_factor = 8
+    low_freq_factor = 1
+    high_freq_factor = 4
+    old_context_len = 8192  # original llama3 length
+
+    low_freq_wavelen = old_context_len / low_freq_factor
+    high_freq_wavelen = old_context_len / high_freq_factor
+    new_freqs = []
+    for freq in freqs:
+        wavelen = 2 * math.pi / freq
```
Comment (malfet) [inline build/model.py:379]:
```suggestion
        wavelen = 2 * torch.pi / freq
```

File: build/model.py [hunk 2]
```diff
@@ -5,6 +5,8 @@
 # LICENSE file in the root directory of this source tree.
 import json
 import os
+import math
+
```
Comment (malfet) [inline build/model.py:9]: Use pi from torch rather than math
```suggestion

```

File: config/data/models.json [hunk 3]
```diff
@@ -40,6 +40,12 @@
         "distribution_path": "meta-llama/Meta-Llama-3-70B-Instruct",
         "transformer_params_key": "Meta-Llama-3-70B"
     },
+    "meta-llama/Meta-Llama-3.1-8B-Instruct": {
```
Comment (kartikayk) [inline config/data/models.json:43]: nit: technically you can also add support for the pre-trained model ([meta-llama/Meta-Llama-3.1-8B](https://huggingface.co/meta-llama/Meta-Llama-3.1-8B)) and the llama guard ([meta-llama/Llama-Guard-3-8B](https://huggingface.co/meta-llama/Llama-Guard-3-8B)). Not a requirement though

**Review by malfet — APPROVED** [review]

File: build/model.py [hunk 4]
```diff
@@ -361,15 +365,41 @@ def forward(self, x: Tensor) -> Tensor:
...
+def apply_scaling(freqs: torch.Tensor):
+    # Values obtained from grid search
+    scale_factor = 8
+    low_freq_factor = 1
+    high_freq_factor = 4
+    old_context_len = 8192  # original llama3 length
+
+    low_freq_wavelen = old_context_len / low_freq_factor
+    high_freq_wavelen = old_context_len / high_freq_factor
+    new_freqs = []
```
Comment (malfet) [inline build/model.py:377]: Hmm, it feels like one can write this logic in a much more pytorchy-style:
```
new_freqs = 360 / freq.rad2deg()
new_freqs[new_frequs > high_freq_wavelen ] /= scale_factor
new_frequs[new_frequs < low_freq_wavelen * scale_factor] = apply_smooth_here 
```

**Comment by byjlw** [comment]
Awesome! Please update the readme so people can discover it :)

File: config/data/models.json (hunk 3 above)
Comment (orionr) [reply config/data/models.json:43]: Probably want support for the pre-trained (non-Instruct) version to match with our support for Llama 3.0?

Comment (Gasoonjia) [reply config/data/models.json:43]: oh yeah of course. This PR is focused on enabling llama3.1 in torchchat, so didn't cover all possible models. Will have another PR to handle that.

**Pull request merged**